code/modules/mob/living/silicon/say.dm

```diff
@@ -10,24 +10,41 @@
 	if(issilicon(src))
 		var/mob/living/silicon/S = src
 		desig = trim_left(S.designation + " " + S.job)
-	var/message_a = say_quote(message)
-	var/rendered = "Robotic Talk, <span class='name'>[name]</span> <span class='message'>[message_a]</span>"
+	var/large_message_a = say_quote(message, list("robot big"))
+	var/message_a = say_quote(message, list("robot"))
+	var/mob/living/silicon/ai/true_ai_core
+	if(iscyborg(src))  // this detects if a borg is AI shell, so that they can be loud always
+		var/mob/living/silicon/robot/ai_shell = src
+		true_ai_core = ai_shell.mainframe
 	for(var/mob/M in GLOB.player_list)
 		if(M.binarycheck())
 			if(isAI(M))
-				var/renderedAI = "<span class='srt_radio binarysay'>Robotic Talk, <a href='?src=[REF(M)];track=[html_encode(name)]'><span class='name'>[name] ([desig])</span></a> <span class='message'>[message_a]</span></span>"
-				to_chat(M, renderedAI)
+				var/loud = FALSE
+				if(M == src) //AI hears only itself on loud mode.
+					loud = TRUE
+				var/rendered = "<span class='srt_radio binarysay'>Robotic Talk, <a href='?src=[REF(M)];track=[html_encode(name)]'><span class='name'>[name] ([isAI(M) ? "[desig]" : ""])</span></a> [loud ? "[large_message_a]" : "[message_a]"]</span>"
```

Comment:
Uh, what's the point of `([isAI(M) ? "[desig]" : ""])`?? It looks to be a mistake

Reply:
It looks redundant check to me? Should it be another extra check beside line 21?